### PR TITLE
Make sure private projects stay secret for those without access

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Fixes [an issue found in release testing](https://docs.google.com/document/d/1tyTChgLM8ZKCIyP05yDeyzS7y-s2ii5lm5WltREO3so/edit?disco=AAAATE15Opk).

## How did you test this code?

Extended an existing test case. It previously checked only the `projects` endpoint, but we need to have analogous logic in the `Organization` serializer too.

### CODE REVIEW

File: posthog/api/organization.py [hunk 1]
```diff
@@ -132,6 +131,14 @@ def get_setup(self, instance: Organization) -> Dict[str, Union[bool, int, str, N
             "has_invited_team_members": instance.invites.exists() or instance.members.count() > 1,
         }
 
+    def get_teams(self, instance: Organization) -> List[Dict[str, Any]]:
+        visible_teams = (
```
Comment (paolodamico) [inline posthog/api/organization.py:135]: This seems inefficient and it's an endpoint we call on every app load, I think we could accomplish this with a single query. We could have a similar method to `get_effective_membership_level` that already appends the effective level to each team in the list.

**Review by paolodamico — CHANGES_REQUESTED** [review]

**Comment by paolodamico** [comment]
Also FYI tested functionality and it does work as expected.

File: posthog/api/organization.py (hunk 1 above)
Comment (Twixes) [reply posthog/api/organization.py]: It's still inefficient but now a bit less so indeed, by reusing the fact that `effective_membership_level` has to be serialized anyway.

**Review by paolodamico — APPROVED** [review]
Probably should be optimized further, but I doubt it would make an immediate difference, ship it!

**Pull request merged**